tests/ui/UnreadIndicatorsTest.js

```diff
@@ -0,0 +1,521 @@
+    // fetch() never gets called so it does not need mocking) or we might have fetch throw an error to test error handling
+    // behavior. But here we just want to treat all API requests as a generic "success" and in the cases where we need to
+    // simulate data arriving we will just set it into Onyx directly with Onyx.merge() or Onyx.set() etc.
+    global.fetch = TestHelper.getGlobalFetchMock();
+
+    // We need a bit more time for this test in some places
+    Linking.setInitialURL('https://new.expensify.com/r/1');
+    appSetup();
+});
+
+/**
+ * @param {RenderAPI} renderedApp
+ */
+function scrollUpToRevealNewMessagesBadge(renderedApp) {
+    fireEvent.scroll(renderedApp.getByA11yLabel('List of chat messages'), {
+        nativeEvent: {
+            contentOffset: {
+                y: 250,
+            },
+            contentSize: {
+                // Dimensions of the scrollable content
+                height: 500,
+                width: 100,
+            },
+            layoutMeasurement: {
+                // Dimensions of the device
+                height: 700,
+                width: 300,
+            },
+        },
+    });
+
+    // We advance the timer since we must wait for the animation to end
+    // and the new style to be reflected
+    jest.advanceTimersByTime(100);
+}
+
+/**
+ * @param {RenderAPI} renderedApp
+ * @return {Boolean}
+ */
+function isNewMessagesBadgeVisible(renderedApp) {
+    const badge = renderedApp.getByA11yHint('Scroll to newest messages');
+    return badge.props.style.transform[0].translateY === 10;
+}
+
+/**
+ * @param {RenderAPI} renderedApp
+ * @return {Promise}
+ */
+function navigateToSidebar(renderedApp) {
+    const reportHeaderBackButton = renderedApp.getByA11yHint('Navigate back to chats list');
+    fireEvent(reportHeaderBackButton, 'press');
+    return waitForPromisesToResolve();
+}
+
+/**
+ * @param {RenderAPI} renderedApp
+ * @param {Number} index
+ * @return {Promise}
+ */
+function navigateToSidebarOption(renderedApp, index) {
+    const optionRows = renderedApp.getAllByA11yHint('Navigates to a chat');
+    fireEvent(optionRows[index], 'press');
+    return waitForPromisesToResolve();
+}
+
+/**
+ * @param {RenderAPI} renderedApp
+ * @return {Boolean}
+ */
+function isDrawerOpen(renderedApp) {
+    const sidebarLinks = renderedApp.queryAllByA11yLabel('List of chats');
+    return !lodashGet(sidebarLinks, [0, 'props', 'accessibilityElementsHidden']);
+}
+
+const REPORT_ID = 1;
+const USER_A_ACCOUNT_ID = 1;
+const USER_A_EMAIL = 'user_a@test.com';
+const USER_B_ACCOUNT_ID = 2;
+const USER_B_EMAIL = 'user_b@test.com';
+const USER_C_ACCOUNT_ID = 3;
+const USER_C_EMAIL = 'user_c@test.com';
+
+/**
+ * Sets up a test with a logged in user that has one unread chat from another user. Returns the <App/> test instance.
+ *
+ * @returns {RenderAPI}
+ */
+function signInAndGetAppWithUnreadChat() {
```

Comment:
Great point. I think there's also a chance that a lot of UI tests could not need this since they might test more isolated individual components and not flows that require navigation. I think this would be good to add in a future PR though once there is a second test that needs this behavior.